### PR TITLE
[Content] Overviews/BufNMF

### PR DESCRIPTION
**Pull request by tedmoore**
@jamesb93 It's super slick to put content in here and get it up and rolling. Love it.

Two questions:
1. Is there a way to make the images a different size, specifically larger?
2. Also, is there a way to not have my mp3's gross filename appear on the webpage?

(also, n.b., since this content has already been reviewed, I'm not merging yet just for these two questions, not per the content, but of course if anyone sees any more content suggestions, always happy to hear them!)

Thank you!

### CODE REVIEW

**Comment by jamesb93** [comment]
> @jamesb93 It's super slick to put content in here and get it up and rolling. Love it.
> 

Thanks man, appreciate the kind feedback :)

> 1. Is there a way to make the images a different size, specifically larger?

How larger you wanna go? I can see wanting to have them bleed to the edge of the main content column (basically to the width of the text). If you want them to bleed outside the content then that's another thing to discuss and a PR for me to make. We can do it if you reckon its valuable though.

> 2. Also, is there a way to not have my mp3's gross filename appear on the webpage?
> 

Definitely. I'll update the `<Audio />` component to have a flag you can pass.

```sv
<Audio 
showTitle={false}
/>
```

or something. We can do it in an adjacent PR or one after merge. This is something I've wanted to mix up aesthetically anyway.

> (also, n.b., since this content has already been reviewed, I'm not merging yet just for these two questions, not per the content, but of course if anyone sees any more content suggestions, always happy to hear them!)

I would like to take just one more chance to read the thing in the morning if that's okay? Some things I can spot I'll put in my review.

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:15]: Because the title is already rendered in the layout (Audio Decomposition using BufNMF) it is necessary to make your first tag (Decomposition) here a `<h2>` or `##`. This also improves accessibility.

image

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:20]: I'll need to check this but we should compress the life out of this till its as small as possible. My usual settings are 128kbps with the `ffmpeg` or `REAPER` converter. For comparison the oboe sample is 5 times as long but about the same size currently. Smaller is better for making the page __fast__.

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:38]: This should be all lower (non-negative matrix factorisation) as well as british english.

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:44]: If it is short enough we can put it into a CodeBlock. Your call.

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:134]: If there were any other resources you wanted to link I've been using a <ResourceLink /> component under a `## Related Resources` header at the bottom. You can see the example rendered here: https://learn.flucoma.org/reference/umap

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:6]: i think we could add some more tags here:

- nmf
- analysis
- machine learning

More tags helps the search too :)

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:51]: I think that all images should have labels. They look floaty without them and it helps screen readers (because it provides the alt tag).

**Review by jamesb93 — CHANGES_REQUESTED** [review]
Overall great. I've added some requested changes to do with some code enforcement stuff (providing labels to all images), as well as taking down the first header to a 2 rather than 1. Overall these are very fast changes and nothing hugely structural.

File: src/routes/overviews/bufnmf.svx
Comment (tedmoore) [reply src/routes/overviews/bufnmf.svx:44]: I'd rather leave it as a download so that all three CCEs are equally represented, not like, see this SC code block, but you others gotta download.

Comment (jamesb93) [reply src/routes/overviews/bufnmf.svx:44]: Slummin' it in Max and PD 😂

**Comment by tedmoore** [comment]
> How larger you wanna go? I can see wanting to have them bleed to the edge of the main content column (basically to the width of the text). If you want them to bleed outside the content then that's another thing to discuss and a PR for me to make. We can do it if you reckon its valuable though.

I think starting with width of the text is good. If that's easy I'll be curious to see how it looks. Generally I think bigger is better for less squinting. 

On that note, is it possible for the text column itself to be wider? I don't know if there are standards around this stuff but it's rather narrow for my taste.

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [inline src/routes/overviews/bufnmf.svx:4]: Another british english nit pick

**Comment by jamesb93** [comment]
> > How larger you wanna go? I can see wanting to have them bleed to the edge of the main content column (basically to the width of the text). If you want them to bleed outside the content then that's another thing to discuss and a PR for me to make. We can do it if you reckon its valuable though.
> 
> I think starting with width of the text is good. If that's easy I'll be curious to see how it looks. Generally I think bigger is better for less squinting.

Okay, I'll flag it as an issue and I'll make a separate PR for it. For now keep it as is and when this other PR gets merged it'll apply to Images everywhere.

> 
> On that note, is it possible for the text column itself to be wider? I don't know if there are standards around this stuff but it's rather narrow for my taste.

I did a fair bit of research on this to try get it right. What I found was that most UX/UI guides recommend a maximum of 50-60 characters but you can stretch as far as 70. ATM the CSS is set to constrain the text to a maximum of 70 so I'd be cautious on going any wider. There are things we can do to make content bleed outside the main column though which might ameliorate your concerns :P such as using a full bleed layout which distill uses a lot for example: https://distill.pub/2021/understanding-gnns/#challenges

File: src/routes/overviews/bufnmf.svx
Comment (tedmoore) [reply src/routes/overviews/bufnmf.svx:4]: Factorization is with an "s"? FWIW British spellings don't trigger red flags in my mind so I can't say that I'll really be catching them. I'll sure try!!

**Comment by tedmoore** [comment]
> There are things we can do to make content bleed outside the main column though which might ameliorate your concerns :P such as using a full bleed layout which distill uses a lot for example

I like the way distill looks.

File: src/routes/overviews/bufnmf.svx
Comment (jamesb93) [reply src/routes/overviews/bufnmf.svx:4]: No probs :P I'll flag them where I see them and eventually write a test for it :P

**Comment by jamesb93** [comment]
> > There are things we can do to make content bleed outside the main column though which might ameliorate your concerns :P such as using a full bleed layout which distill uses a lot for example
> 
> I like the way distill looks.

Same. It's a little wide for my liking and I find myself tracking back to the start of the line and losing my place a bit but the way they use visuals and widgets is incredible.

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:24]: I think should be type='pointer' for when you note to someone there is a related resource inline

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:40]: kind of a long sentence

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:20]: The layout will already render this for you so you can omit it or start with "introduction" or something else

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:22]: Do you envisage the superscript to be a footnote link?

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:53]: I'd opt for `` instead of **** so it looks like a code thing

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:55]: flags or parameters or attributes? what do ya reckon

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:59]: pointer type again

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:65]: Maybe flag here that the best way to do this is to ensure you use the fftsetting when processing with the family of objects.

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [inline src/routes/reference/bufnmf.svx:70]: I think we should call these Related Resources and use the <ResourceLink /> component. See an example here:

https://github.com/flucoma/learn-website/blob/f491c825f2bdc9f6ad54d36b598ef5b07468e445/src/routes/reference/umap.svx#L42

That way we can make it linkable and update that portion of every document in one place.

File: src/routes/overviews/bufnmf.svx
Comment (tedmoore) [reply src/routes/overviews/bufnmf.svx:4]: A test for British english! That's some machine assistance right there.

File: src/routes/reference/bufnmf.svx
Comment (tedmoore) [reply src/routes/reference/bufnmf.svx:40]: I agree. This was after squeezing in all the suggestions from @tremblap . Now that I read it back in context, fwhew.

File: src/routes/reference/bufnmf.svx
Comment (tedmoore) [reply src/routes/reference/bufnmf.svx:20]: ah i understand.

File: src/routes/reference/bufnmf.svx
Comment (tedmoore) [reply src/routes/reference/bufnmf.svx:22]: oh yes, this was a question i had for you. can i haz footnotes?

File: src/routes/reference/bufnmf.svx
Comment (tedmoore) [reply src/routes/reference/bufnmf.svx:55]: yeah, i guess i'm actually going to go with arguments since that's what the list down below is called and these will be in that list.

File: src/routes/reference/bufnmf.svx
Comment (tedmoore) [reply src/routes/reference/bufnmf.svx:65]: is this a Max specific thing?

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [reply src/routes/reference/bufnmf.svx:22]: Theres a bit of thinking that needs to go into how this gets pulled out the markdown abstract syntax tree and links up automatically but there is some existing libraries that can help me:

https://github.com/remarkjs/remark-gfm

It'll need to be a PR though as it's more than a one day job including exploration 'n' stuff.

File: src/routes/reference/bufnmf.svx
Comment (jamesb93) [reply src/routes/reference/bufnmf.svx:65]: Shouldn't be. Just that if you do some NMF in one place and wait to supply that data to another nmf thing later, being explicit and consistent with that parameter helps to ensure you get the right sizes rather than relying on the default (which should be the same but its probably good hygiene and practice to do otherwise).

File: src/routes/reference/bufnmf.svx
Comment (tedmoore) [reply src/routes/reference/bufnmf.svx:22]: never mind. we'll just roll with the `related resources`

**Comment by jamesb93** [comment]
FYI @tedmoore this will fail to build for the reason that there is a link to a document that doesn't yet exist.

`> 404 /overviews/seeding-bases-and-activations (linked from /reference/bufnmf)`

If that link (seeding-bases-and-activations) is yet to be written make the link `[placeholder]()` and flag an issue or make a note somewhere to update it later.

**Comment by tedmoore** [comment]
Vercel won't let me see the "Details". How can I see what is failing?
vercel

**Comment by jamesb93** [comment]
> Vercel won't let me see the "Details". @james how can I see what is failing?
> vercel

Build locally or make an account are the only options. If you sign in with github it should authenticate you automatically on our team vercel and youll have access to see the logs.

**Pull request merged**
